src/main/webapp/parse.js

```diff
@@ -44,10 +46,13 @@ function sendData() {
             try {
                 const ingrHTML = document.getElementsByTagName("ul").item(0).children;
                 const list = document.getElementsByClassName("mw-parser-output")[0];
+                const ingredients = Array.from(ingrHTML).map(el => el.innerText);
                 const descriptionHTML = list.getElementsByTagName("p")[1];
+                var description = "";
+                if (descriptionHTML != null) {
+                    description = descriptionHTML.textContent ?? "";
+                }
```

Comment:
A more idiomatic way of writing this is
```
const description = descriptionHTML == null ? : L descriptionHtml.textContent ?? "";
```
or, in a more modern way (not supported in older browsers, but I dont think we care)
```
const description = descriptionHTML?.descriptionHtml.textContent ?? "";
```
The `?.` is call an optional chain operator: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

Either way, in general it's preferred not to create variables with temporary values just to reassign them in an if, e.g.
```
var x = "default_value";
if (some_condition) {
    x = "non_default_value;
}
```
It's more common to assign the value to the result of an inline `if`, in any form (`?`, `||`, `&&`, `?.`), e.g.
```
const x = some_condition ? "non_default_value" : "default_value";
```